code/game/gamemodes/cult/runes.dm

```diff
@@ -1,1053 +1,518 @@
+/obj/effect/rune/malformed
+	cultist_name = "malformed rune"
+	cultist_desc = "A senseless rune written in gibberish. No good can come from invoking this."
+	invocation = "Ra'sha yoka!"
 
-	if(ticker.mode.name == "cult")
-		ticker.mode:add_cultist(D.mind)
-	else
-		ticker.mode.cult+=D.mind
-
-	D.mind.special_role = "Cultist"
-	D << "<font color=\"purple\"><b><i>Your blood pulses. Your head throbs. The world goes red. All at once you are aware of a horrible, horrible truth. The veil of reality has been ripped away and in the festering wound left behind something sinister takes root.</b></i></font>"
-	D << "<font color=\"purple\"><b><i>Assist your new compatriots in their dark dealings. Their goal is yours, and yours is theirs. You serve the Dark One above all else. Bring It back.</b></i></font>"
-
-	var/mob/living/user = usr
-	while(this_rune && user && user.stat==CONSCIOUS && user.client && user.loc==this_rune.loc)
-		user.take_organ_damage(1, 0)
-		sleep(30)
-	if(D)
-		D.visible_message("<span class='danger'>[D] slowly dissipates into dust and bones.</span>", \
-		"<span class='danger'>You feel pain, as bonds formed between your soul and this homunculus break.</span>", \
-		"<span class='italics'>You hear faint rustle.</span>")
-		D.dust()
-	return
+/obj/effect/rune/malformed/New()
+	..()
+	icon_state = "[rand(1,6)]"
+	color = rgb(rand(0,255), rand(0,255), rand(0,255))
 
+/obj/effect/rune/malformed/invoke(mob/living/user)
+	user << "<span class='warning'><b>You feel your life force draining. The Geometer is displeased.</b></span>"
+	user.apply_damage(30, BRUTE)
+	qdel(src)
 
-///Imbue Talisman: Put a compatible rune near this one and a paper on top of the imbue rune. Activate, and the compatible rune will be imbued into the paper.
+var/list/teleport_runes = list()
+//Rite of Translocation: Warps the user to a random teleport rune with the same keyword.
+/obj/effect/rune/teleport
+	cultist_name = "Rite of Translocation"
+	cultist_desc = "Warps the user to a random rune of the same keyword."
+	invocation = "Sas'so c'arta forbici!"
+	icon_state = "2"
+	color = rgb(0, 0, 255)
+	var/keyword = "veri"
+
+/obj/effect/rune/teleport/New()
+	..()
+	teleport_runes.Add(src)
+
+/obj/effect/rune/teleport/Destroy()
+	teleport_runes.Remove(src)
+	..()
+
+/obj/effect/rune/teleport/invoke(mob/living/user)
+	var/list/potential_runes = list()
+	for(var/obj/effect/rune/teleport/T in teleport_runes)
+		if(T.keyword == src.keyword && T != src)
+			potential_runes.Add(T)
+
+	if(!potential_runes)
+		fail_invoke()
+		log_game("Teleport rune failed - no candidates with matching keyword")
+		return
 
+	var/obj/effect/rune/selected_rune = pick(potential_runes)
+	user.visible_message("<span class='warning'>[user] vanishes in a flash of red light!</span>", \
+						 "<span class='danger'>Your vision blurs, and you suddenly appear somewhere else.</span>")
+	user.loc = get_turf(selected_rune)

```

Comment:
```user.Move(get_turf(selected_rune))``` or ```user.forceMove(get_turf(selected_rune))``` (I believe both handle everything)
directly setting loc will fail to trigger area based stuff and other junk.